File: src/types/system.ts
export interface SystemInfo {
  hostname: string;
  os_version: string; // backend-only
  cpu_usage: number;
  ram_used_bytes: number;
  ram_total_bytes: number;
  uptime_seconds: number;
}

export interface DiskUsage {
  name: string; // backend-only
  mount_point: string; // backend-only
  total_bytes: number;
  used_bytes: number;
  available_bytes: number; // backend-only
  filesystem: string; // backend-only
}

File: src-tauri/src/models/system.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SystemInfo {
    pub hostname: String,
    pub os_version: String,
    pub cpu_usage: f32,
    pub ram_used_bytes: u64,
    pub ram_total_bytes: u64,
    pub uptime_seconds: u64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DiskUsage {
    pub name: String,
    pub mount_point: String,
    pub total_bytes: u64,
    pub used_bytes: u64,
    pub available_bytes: u64,
    pub filesystem: String,
}
